OSOE-593: Use <ExecWithMutex> only on Windows

0liver opened the pull request:
[OSOE-593](https://lombiq.atlassian.net/browse/OSOE-593)

[OSOE-593]: https://lombiq.atlassian.net/browse/OSOE-593?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
Use <ExecWithMutex> only on Windows

Commit:
Refactor to fix Linux error

Commit:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-593

Commit:
Update NE nuget package to latest alpha

Commit:
Make print-config-summary an optional parameter

dministro committed:
Targeting build-and-test.yml on issue branch

dministro committed:
Updating submodule:
 - Lombiq.GitHub.Actions: Fixing default value

dministro commented on .github/workflows/build-and-test.yml:
```diff
@@ -11,7 +11,7 @@ jobs:
     if: github.ref_name != github.event.repository.default_branch
     name: Build and Test - root solution (larger runners)
-    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@dev
+    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@issue/OSOE-593
```
Next time if you made changes in LGHA don't forget to target the issue branch.

See: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/4757787642

0liver replied:
I will try hard!

dministro committed:
Targeting dev in build-and-test.yml
Updating submodules:
 - Lombiq.GitHub.Actions: fast-forward
 - Lombiq.NodeJs.Extensions: fast-forward
